gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/x509/X509AuthenticationUserDetailsService.groovy

```diff
@@ -62,38 +83,64 @@ class X509AuthenticationUserDetailsService implements AuthenticationUserDetailsS
 
     def x509 = (X509Certificate) token.credentials
 
-    def email
+    String email
     if (userIdentifierExtractor) {
       email = userIdentifierExtractor.fromCertificate(x509)
     }
     if (email == null) {
-      email = emailFromSubjectAlternativeName(x509) ?: token.principal
+      email = emailFromSubjectAlternativeName(x509) ?: token.principal?.toString()
     }
 
-    def roles = []
-    if (email) {
-      log.debug("Adding email {} to roles.", email)
-      roles.add(email)
+    if (email == null) {
+      return null
     }
 
-    String username = email as String
-    if (rolesExtractor) {
-      def extractedRoles = rolesExtractor.fromCertificate(x509)
-      log.debug("Extracted roles {}", extractedRoles)
-      roles += extractedRoles
-      permissionService.loginWithRoles(username, roles)
-    } else {
-      permissionService.login(username)
-    }
+    def roles = handleLogin(email, x509)
 
     log.debug("Roles we have now: {}", roles)
     return new User(
         email: email,
-        allowedAccounts: allowedAccountsSupport.filterAllowedAccounts(username, roles),
+        allowedAccounts: allowedAccountsSupport.filterAllowedAccounts(email, roles),
         roles: roles
     )
   }
 
+  @PackageScope
+  Collection<String> handleLogin(String email, X509Certificate x509) {
+
+    final long now = clock.millis()
+    final boolean loginDebounceEnabled = dynamicConfigService.isEnabled('x509.loginDebounce', false)
```

Comment:
Where the did the relative priority of loading fast properties vs from yaml end up at?

Is the `DynamicConfigService` going to favor something defined in yaml?

Reply:
it reads directly from the environment so whatever priority order is applied there (CLI args take priority over JVM args take priority over ENV variables, take priority over YAML)

In our case our remote dynamic property implementation gets added as highest priority once it is loaded so it would win over all those

Reply:
Sorry to come here for a long time. But how do I enable it? I set up it in many ways and it is still not working.